spec/rspec/rails/matchers/have_http_status_spec.rb

```diff
@@ -177,210 +177,74 @@ def create_response(opts = {})
     end
   end
 
-  context "with general status code group", ":error" do
-    # The error query is an alias for `server_error?`:
-    #
-    # - https://github.com/rails/rails/blob/ca200378/actionpack/lib/action_dispatch/testing/test_response.rb#L27
-    # - https://github.com/rails/rails/blob/master/actionpack/lib/action_dispatch/testing/test_response.rb
-    #
-    # `server_error?` is part of the Rack Helpers and is defined as:
-    #
-    #     status >= 500 && status < 600
-    #
-    # See:
-    #
-    # - https://github.com/rack/rack/blob/ce4a3959/lib/rack/response.rb#L122
-    # - https://github.com/rack/rack/blob/master/lib/rack/response.rb
-
-    subject(:have_error_status) { have_http_status(:error) }
+  shared_examples_for "response statuses" do |code, http_status_symbol|
+    let(:http_status_result) { have_http_status(http_status_symbol)}
 
-    it_behaves_like "supports different response instances" do
-      subject(:matcher) { have_error_status }
-
-      let(:code) { 555 }
+    it "returns true for a response with a #{code} status code" do
+      response = create_response(:status => code)
+      expect( http_status_result.matches?(response) ).to be(true)
     end
 
-    describe "matching a response" do
-      it "returns true for a response with a 5xx status code" do
-        any_5xx_code = 555
-        response     = create_response(:status => any_5xx_code)
-
-        expect( have_error_status.matches?(response) ).to be(true)
-      end
-
-      it "returns false for a response with a different code" do
-        client_error_code = 400
-        response          = create_response(:status => client_error_code)
-
-        expect( have_error_status.matches?(response) ).to be(false)
+    it "returns true for a response which starts with the same number (i.e. 4XX and 4ZZ) " do
+      if code == 404
```

Comment:
We don't want if's inside our tests, we like the seperate tests, you've not removed duplication here.